src/tool_doswin.c

```diff
@@ -763,6 +763,7 @@ static void init_terminal(void)
 }
 
 /* clean possible sensitive information from the win32 command-line */
+#ifdef NDEBUG
```

Comment:
A `NDEBUG` does not mean a *Release Build*. 
This better test for `#if !defined(_DEBUG)` instead.

Reply:
> A `NDEBUG` does not mean a _Release Build_. This better test for `#if !defined(_DEBUG)` instead.

I'm not sure about this. I think `NDBUG` is *standard* and disables debug assertions. So this is expected to be defined for "release" builds. `_DEBUG` is *MSVC-specicifc* and is defined by MSVC when linking against the "debug" CRT (option `/MDd` or `/Mtd`). Is `_DEBUG` set by (or does it have any significance for) other compilers, like MinGW/GCC or Clang?

https://learn.microsoft.com/en-us/cpp/build/reference/md-mt-ld-use-run-time-library?view=msvc-170#remarks

> Do other windows command line programs "normally" wipe the command line like this?

Well, I cannot give you a concrete example. But from the [*The Old New Thing* article](https://devblogs.microsoft.com/oldnewthing/20091125-00/?p=15923) it should be clear that even though "snooping" another process' command-line is possible, you really shouldn't rely on it:

> Remember that from Win32’s point of view, the command line is just a string that is copied into the address space of the new process. How the launching process and the new process interpret this string is governed not by rules but by convention.

> What’s more, since the string is merely a “preinitialized variable”, a process could in principle (and many do in practice) write to the memory that holds the command line, in which case, if you go snooping around for it, you’ll see the modified command line. There is **no** secret hiding place where the kernel keeps the “real original command line”

> What does this mean for people who disregard this principle and go after the command line of another process? You have to understand what you are getting is non-authoritative information. In fact, it’s worse. It’s information the application itself may have changed in order to try to fool you, so do **not** use it to make important decisions.

Reply:
>  I think NDEBUG is standard and disables debug assertions. 

Disabling `assert()` is madness even for *Release mode* IMHO.

>  Is _DEBUG set by (or does it have any significance for) other compilers, like MinGW/GCC or Clang

`clang-cl` has it (`-MDd` or `-MTd`). MinGW does not have `_DEBUG` mode. Hence `!defined(_DEBUG)` is TRUE.
But why do you want `clean_cmdln()` only for *Release mode*?

Reply:
...because otherwise some of the checks fail 😞